xskillscore/core/probabilistic.py

```diff
@@ -485,3 +489,123 @@ def discrimination(
+            (True or 1 for event occurance, False or 0 for non-occurance).
+        forecasts : xarray.Dataset or xarray.DataArray
+            The forecast likelihoods of the event. Data should be between 0 and 1.
+        dim : str or list of str, optional
+            Dimension(s) over which to compute the histograms
+            Defaults to None meaning compute over all dimensions.
+        probability_bin_edges : array_like, optional
+            Probability bin edges used to compute the reliability. Bins include the left most edge, \
+            but not the right. Defaults to 6 equally spaced edges between 0 and 1+1e-8
+        keep_attrs : bool
+            If True, the attributes (attrs) will be copied from the first input to the new one.
+            If False (default), the new object will be returned without attributes.
+
+        Returns
+        -------
+        xarray.Dataset or xarray.DataArray
+            The relative frequency of occurrence for each probability bin
+        xarray.Dataset or xarray.DataArray
+            The sample size of each probability bin
+
+        Examples
+        --------
+        >>> forecasts = xr.DataArray(np.random.normal(size=(3,3,3)),
+        ...                          coords=[('x', np.arange(3)),
+        ...                                  ('y', np.arange(3)),
+        ...                                  ('ensemble', np.arange(3))])
+        >>> observations = xr.DataArray(np.random.normal(size=(3,3)),
+        ...                            coords=[('x', np.arange(3)),
+        ...                                    ('y', np.arange(3))])
+        >>> rel, samples = reliability(observations > 0.1, (forecasts > 0.1).mean('ensemble'), dim='x')
+
+        Notes
+        -----
+        See http://www.cawcr.gov.au/projects/verification/
+    """
+
+    def _reliability(o, f, bin_edges):
+        """Return the reliability and number of samples per bin
+        """
+        # I couldn't get dask='parallelized' working in this case so dealing with dask arrays explicitly
+        is_dask_array = isinstance(o, darray.core.Array) | isinstance(
+            f, darray.core.Array
+        )
+
+        if is_dask_array:
+            r = []
+            N = []
+        else:
+            r = np.zeros((*o.shape[:-1], len(bin_edges) - 1), dtype=float)
+            N = np.zeros_like(r)
+
+        for i in range(len(bin_edges) - 1):
+            # Follow xhistogram: all bins are half-open
+            # https://github.com/xgcm/xhistogram/issues/18
+            f_in_bin = (f >= bin_edges[i]) & (f < bin_edges[i + 1])
+            o_f_in_bin = o & f_in_bin
+            N_f_in_bin = f_in_bin.sum(axis=-1)
+            N_o_f_in_bin = o_f_in_bin.sum(axis=-1)
+            if is_dask_array:
+                r.append(N_o_f_in_bin / N_f_in_bin)
+                N.append(N_f_in_bin)
+            else:
+                r[..., i] = N_o_f_in_bin / N_f_in_bin
```

Comment:
I get this warning here:
`Users/aaron.spring/Coding/xskillscore/xskillscore/core/probabilistic.py:570: RuntimeWarning: invalid value encountered in long_scalars
  r[..., i] = N_o_f_in_bin / N_f_in_bin`

Reply:
Yeah, this happens when there are no samples in a particular bin, leading to division by 0.